IBAnimatable/ExplodeAnimator.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ExplodeAnimator.swift
+//  IBAnimatableApp
+//
+//  Created by Tom Baranes on 03/04/16.
+//  Copyright © 2016 Jake Lin. All rights reserved.
+//
+
+import UIKit
+
+public class ExplodeAnimator: NSObject, AnimatedTransitioning {
+  // MARK: - AnimatorProtocol
+  public var transitionAnimationType: TransitionAnimationType
+  public var transitionDuration: Duration = defaultTransitionDuration
+  public var reverseAnimationType: TransitionAnimationType?
+  public var interactiveGestureType: InteractiveGestureType?
+  
+  init(transitionDuration: Duration) {
+    self.transitionDuration = transitionDuration
+    self.transitionAnimationType = .Explode
+    self.reverseAnimationType = .Explode
+    super.init()
+  }
+}
+
+extension ExplodeAnimator: UIViewControllerAnimatedTransitioning {
+  public func transitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval {
+    return retrieveTransitionDuration(transitionContext)
+  }
+  
+  public func animateTransition(transitionContext: UIViewControllerContextTransitioning) {
+    let (tempfromView, tempToView, tempContainerView) = retrieveViews(transitionContext)
+    guard let fromView = tempfromView, toView = tempToView, containerView = tempContainerView else {
+      transitionContext.completeTransition(true)
+      return
+    }
+    
+    containerView.addSubview(toView)
+    containerView.sendSubviewToBack(toView)
+    
+    let snapshots = createSnapshots(toView: toView, fromView: fromView, containerView: containerView)
+    containerView.sendSubviewToBack(fromView)
+    animateSnapshotsExplode(snapshots) {
+      transitionContext.completeTransition(!transitionContext.transitionWasCancelled())
+    }
+  }
+  
+}
+
+private extension ExplodeAnimator {
+  
+  func createSnapshots(toView toView: UIView, fromView: UIView, containerView: UIView) -> [UIView] {
+    let size = toView.frame.size
+    var snapshots = [UIView]()
+    let xFactor: CGFloat = 10.0
+    let yFactor = xFactor * size.height / size.width
+    
+    let fromViewSnapshot = fromView.snapshotViewAfterScreenUpdates(false)
+    for x in 0.stride(to: Int(size.width), by: Int(size.width / xFactor)) {
```

Comment:
Shall we use `Double(size.width)` instead of `Int(size.width)` because `CGFloat` is more close to `Double` although width or height should not have decimal points.